iscsi/iscsi.go

```diff
@@ -0,0 +1,372 @@
+// Copyright 2017 The Prometheus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package iscsi
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+// for default sysfs mount point
+var (
+	sysPath = "/sys"
```

Comment:
This should not be declared here.  Please take a look at the XFS and bcache collectors.  All of the filenames and access is declared in `sysfs/fs.go`.

Reply:
hi @SuperQ if you look at my update in sysfs/fs.go my ISCSIStats call is basically the same as BcacheStats. This default sysfs is only here for fall back in case someone using it without going from the sysfs/fs.go directly. There are two part of sysfs I need to look into for some iscsi backstore. e.g. ceph, more information need to be match, instead of reading it directly from sysfs's structure.

Reply:
This is generating a very un-clean interface.  Please follow the conventions of the other collectors.